add s3 serving documents (#1015)
pypa/bandersnatch: docs/storage_options.md

S3 Bandersnatch mirrors are designed to be served with s3 static sites and
can also be used with the Amazon CDN service or another CDN service.

> I assume you have already set up an AWS account and S3 bucket, and the Bandersnatch sync job has been proceeded.

Reviewer: ```suggestion
I assume you have already set up an AWS account and S3 bucket, and the Bandersnatch sync job has successfully ran.
```

> ### Enabling website hosting for the bucket
> When you enable the website hosting for a bucket, this bucket can be viewed as static website. Using the s3 domain or your customized domain.

Reviewer: NIT - Markdown wants a \n between all headings - Please do to others.

- I need to add prettier auto formattting to our .md files ...

```suggestion
### Enabling website hosting for the bucket

When you enable the website hosting for a bucket, this bucket can be viewed as static website. Using the s3 domain or your customized domain.
```

> Please read Amazon documents to get detailed instructions: https://docs.aws.amazon.com/AmazonS3/latest/userguide/EnableWebsiteHosting.html

Reviewer: We could make this links ...
```suggestion
Please read Amazon documents to get [detailed instructions](https://docs.aws.amazon.com/AmazonS3/latest/userguide/EnableWebsiteHosting.html)
```

> Most cloud provider who provide s3-compatible service will provide this service as well. Please consult to your service assistant to get detailed instructions.

Reviewer: ```suggestion
Most cloud provider who provide a s3-compatible service will provide this service as well. Please consult to your service assistant to get detailed instructions.
```

### Use CloudFront or other cdn service to speed up the static mirror(optional)
If your mirror is targeted to global clients, you can use CloudFront or other CDN service to speed up the mirror.

> Please read Amazon documents to get detailed instructions: https://docs.aws.amazon.com/AmazonS3/latest/userguide/website-hosting-cloudfront-walkthrough.html

Reviewer: ```suggestion
Please read Amazon documents to get [detailed instructions](https://docs.aws.amazon.com/AmazonS3/latest/userguide/website-hosting-cloudfront-walkthrough.html)
```

### Set redirect or url rewrite in CloudFront or other cdn(optional)
In most cases, packages and index pages are all inside ``/my-s3-bucket/prefix/web``, if you set up a steps above, you should be able to use the mirror like this:

```shell
pip install -i my-s3-bucket.cloudfront.net/prefix/web/simple install django
```

But there are two main disadvantages:
1. The url is quite long and exposing the structure of bucket.
2. Users will be able to view all content in the bucket, including bandersnatch todo file and status file.

It is strongly recommended to set redirect or url rewrite for CDN. Please contact your service assistant for detailed instructions.

## SwiftStack Storage

> To enable SwiftStack support the optional `swift` install must be done:

Reviewer: Thanks! haha
